lib/plugin.js

```diff
@@ -82,55 +106,59 @@ Plugin.prototype.exec = function (server, cb) {
       clearTimeout(timer)
     }
 
-    cb(err)
+    callback(execErr)
   }
 
   if (this.timeout > 0) {
     debug('setting up timeout', name, this.timeout)
     timer = setTimeout(function () {
       debug('timed out', name)
       timer = null
-      const err = new AVV_ERR_READY_TIMEOUT(name)
-      err.fn = func
-      done(err)
+      const readyTimeoutErr = new AVV_ERR_READY_TIMEOUT(name)
+      readyTimeoutErr.fn = func
```

Comment:
I think the only reason we store func as fn to the instance of the timeoutErr, is that we check in plugin-timeout.test.js if the right function was the reason of the error. 
It has imho no real benefit. 

Maybe we should just remove this fn assignment.

Reply:
Add a TODO and possibly do it in a separate PR. In case we break something it's easier to revert.

Reply:
done